.github/workflows/publish.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish
+on:
+  push:
+    branches:
+      - "main"
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  publish:
+    name: Publish NPM package using changesets
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8.1
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: 'pnpm'
+
+      - run: pnpm install --frozen-lockfile
+
+      - name: Create release pull request or publish
+        id: changesets
+        uses: changesets/action@v1
+        with:
+          publish: pnpm run build
```

Comment:
Not quite. This job will detect whether there was a change set generated (it resembles `newsfragment` in `nucypher`) and if there was any, it will create a new PR with a proposed release. This is why this job is called `Create release pull request or publish`.

`changesets` are not fully configured yet, I will resort to manual publishing for now.

Reply:
👍 - gotcha!

Reply:
Why not just call it `Create release pull request`? or `Create release pull request for publish`?

Reply:
Because when this action runs, there are two possible outcomes:
1) It finds a change set and so it creates a PR with a proposed release
2) It detects that PR from 1) was just merged so it publishes the new package to npm
Please see the [action README for details](https://github.com/changesets/action#changesets-release-action)

Reply:
Thanks for link. That makes sense now.